server/controllers/tasks/lib/find_authors_with_isbns_in_wikipedia_articles.js

```diff
@@ -0,0 +1,38 @@
+import { flatMap } from 'lodash-es'
+import { getEntitiesByClaim } from '#controllers/entities/lib/entities'
+import { getMostRelevantWikipediaArticles, matchLabelsInArticle } from '#controllers/entities/lib/get_occurrences_from_external_sources'
+
+export async function findAuthorsWithIsbnsInWikipediaArticles (worksData, authors) {
+  // worksData is built with getAuthorWorksData
+  const { langs, worksUris } = worksData
+  const editions = await Promise.all(getEditionsFromWorks(worksUris))
+    .then(flatMap)
+  const isbns = getIsbnsClaimValues(editions)
+  if (isbns.length === 0) return
+  return authors.find(hasIsbnInWikipediaArticles(langs, isbns))
+}
+
+function getIsbnsClaimValues (editions) {
+  const isbns = []
+  editions.forEach(edition => {
+    const isbn = edition.claims['wdt:P212']?.[0]
+    if (isbn) isbns.push(isbn)
+  })
+  return isbns
+}
+
+const hasIsbnInWikipediaArticles = (langs, isbns) => async suggestion => {
+  const articles = await getMostRelevantWikipediaArticles(suggestion, langs)
+  if (articles.length === 0) return
+  const matchedArticle = articles.find(hasMatchingLabels(isbns))
```

Comment:
This would ignore non-hyphenated or badly hyphenated ISBNs, which unfortunately fairly common.

This could be worked around by using a `findIsbns` functions comparable to what we have on the client (could be added to `server/lib/isbn/isbn.js`):
```js
export const findIsbns = text => text.match(/(97(8|9))?[\d-]{9,14}([\dX])/g)
```
The isbns found in the article could then be passed through the `normalizeIsbn` function, and compared to `isbnsFromClaims.map(normalizeIsbn)`

Reply:
done in f45a88d13

Reply:
The `Isbn` regex misses the `g` flag, so it can not work: you get a signle matching result array instead of a an array of matches. Ex:
```js
"9782203202283 9782749307015".match(/(97(8|9))?[\d-]{9,14}([\dX])/g)
// => [ '9782203202283', '9782749307015' ]
"9782203202283 9782749307015".match(/(97(8|9))?[\d-]{9,14}([\dX])/)
// => [ '9782203202283', '978', '8', '3', index: 0, input: '9782203202283 9782749307015', groups: undefined ]
```

Then, you can not have the `g` flagged regex as a shared regex in `lib/regex`, as you would risk entering [one of the darkest corners of the JS RegExp implementation](https://stackoverflow.com/questions/1520800/why-does-a-regexp-with-global-flag-give-wrong-results) (see 7e21472773f76e730cd56405d05124ffa77ebd01), thus my suggestion to generate a regex object at every function call.

Reply:
done in 2c670fa9d